### DISCUSSION TITLE
ZNJLBL01LM (Aqara Roller shade driver E1) latest firmware

### DISCUSSION THREAD

**Discussion by z-master42** [Koenkk/zigbee2mqtt#24031] [Devices]
Does anyone know where to get the latest firmware?

I have two curtain motors, one is displayed on the OTA page with version 10-09-2022 (in status version 3870), the other with 09-12-2023 (in status with version 3871).

Via Z2M 3870 is the latest one.

I have the feeling that the feedback of the curtain position is more accurate/faster with 3871 than with 3870.

**Comment by davecas49** [comment]
I have 4 of these.  2 are on 3870 and work perfectly.  2 have somehow updated to 3871, and now these two do not expose the cuurent_position attribute.  I am desperately trying to roll back.  Beware of 3871...

**Reply by bkonings** [reply]
Have you found a way to downgrade the firmware?

**Reply by z-master42** [reply]
I also have this attribute with the newer firmware.

**Reply by bkonings** [reply]
It seems 3871 was pulled back:

{
    "battery": 89,
    "charging_status": false,
    "linkquality": 30,
    "motor_speed": "high",
    "position": 0,
    "state": "CLOSE",
    "update": {
        "installed_version": 3871,
        **"latest_version": 3870,**
        "state": "idle"
    },
    "update_available": true,
    "device_temperature": null,
    "motor_state": null,
    "running": null
}

**Reply by z-master42** [reply]
3870 is the latest version available via Z2M. The OTA via Z2M are local and do not run via the device manufacturer. 3871 was on the device when it was delivered.

**Reply by bkonings** [reply]
My ZNJLBL01LM was reporting position and open/closed just fine until somewhere last month.
It is reporting the state correctly after a press on the Stop button. The most logical explanation is that my device updated as some point last month and there is a bug in the new firmware. If that is not it then zigbee2mqtt is broken with the device since somewhere last month (prior to version 2.0.0)

**Reply by z-master42** [reply]
As far as I know, there is no auto-update in Z2M.

**Reply by bkonings** [reply]
> As far as I know, there is no auto-update in Z2M.

No but there is a highly skilled at clicking update user in my instance, me;)

**Comment by Gr4ffy** [comment]
Hi,
I have old firware on ZNJLBL01LM
_installed_version: 3611_

Is it safe upgrade to "latest_version: 3872"?

**Reply by z-master42** [reply]
I haven't had any problems.

**Reply by Gr4ffy** [reply]
Thx, everything OK after upgrade.

**Reply by andyleemeuk** [reply]
I seemed to lose battery percentage reporting when moving from 3611 to 3872

**Reply by z-master42** [reply]
Can't confirm.

**Reply by andyleemeuk** [reply]
<img width="980" height="430" alt="Screenshot from 2026-09-04 17-46-58" src="https://github.com/user-attachments/assets/354882ef-a256-4b62-ae6a-782915895013" />
Both devices compared
<img width="718" height="550" alt="Screenshot from 2026-09-04 17-47-40" src="https://github.com/user-attachments/assets/61572c95-8a39-479d-8979-457ef095967d" />
3872 version

**Reply by andyleemeuk** [reply]
The charge status is populated but the battery amount shows "?" as opposed to the battery level.

**Reply by bkonings** [reply]
After pressing interview I see the same, before it was reporting a battery level that already seemed stuck to me, I had just charged the device and it remained at 78%.

**Reply by z-master42** [reply]
It's all there.
![Screenshot_20260904-192208.png](https://github.com/user-attachments/assets/b5fca673-61b9-4c5e-95dd-2459bce7d56c)

What does your device show under status in Z2M?

**Reply by andyleemeuk** [reply]
{
    "battery": 100,
    "charging_status": false,
    "device_temperature": 21,
    "linkquality": 152,
    "motor_speed": "medium",
    "motor_state": "stopped",
    "position": 100,
    "power_outage_count": 0,
    "running": false,
    "state": "OPEN",
    "update": {
        "installed_version": 3872,
        "latest_release_notes": null,
        "latest_source": null,
        "latest_version": 3872,
        "state": "idle"
    }
}

Which isn't possible as the blind has been used a few dozen times and it is still showing battery as 100%.

**Reply by andyleemeuk** [reply]
<img width="1867" height="190" alt="Screenshot from 2026-09-04 18-36-14" src="https://github.com/user-attachments/assets/b5391179-856d-4aed-be8b-20d957f7e557" />

**Comment by karot9100** [comment]
Hey i have issues with this device aswell. It does not expose the motor state, charging state, running state and also when i press the open button the state changes to open but when i press it again before its fully open it goes back to a closed state althought it is open. I use Home Assistant to active the rollershade tho.

**Reply by z-master42** [reply]
That’s the only thing. When you open or close it, the position displayed indicates that it is already fully open or fully closed. The actual position is only displayed once the device has stopped. When I press Open a second time, absolutely nothing happens.

All the other values are there.
<img width="502" height="1260" alt="grafik" src="https://github.com/user-attachments/assets/c5fe31a6-28c1-499b-bfe0-cfb8e0828efd" />

**Reply by karot9100** [reply]
You have more entity's then i do. I would like to focus on the Motor State. For me this is unknown and i also tried rolling back to an older firmware version. Tried binding the reporting value's and adding them, but i get an error for this. I'll start a seperate thread i think.
<img width="459" height="259" alt="20260715_211802" src="https://github.com/user-attachments/assets/60f53e68-dba5-44bf-a102-6bb34664fc13" />
<img width="416" height="807" alt="20260715_211757" src="https://github.com/user-attachments/assets/9688e0ab-70bf-4862-ae57-fda1ec229414" />
<img width="586" height="156" alt="20260715_201454" src="https://github.com/user-attachments/assets/e4558b62-f6a4-4ea1-a548-a18093c9ba6c" />
